### PR TITLE
fix: Correcting the execution of query with the latest values when run shortcut is pressed 

### LINKED ISSUE #7754: [Bug] Pressing cmd+enter within 1-2 sec of typing executes an old query
## Description

On the query pane, if you press cmd+enter within 1-2 sec of typing a query, it executes an old/previous query

### Steps to reproduce the behavior:

1. Go to query pane
2. Write a select SQL query `select * from users where id = 1`
3. Change the value of id and quickly press cmd+run
4. Notice that the results from the previous value of id are being returned

### Important Details
- Environment: production

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

Pressing cmd+enter while continuously typing the query executes the query with the stale value updated in the redux store.

Fixes #7754 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Type in the query in the query editor of the datasource and then while typing the query continuously press CMD+ENTER to execute the query. After the fix the query should execute with the latest value in the editor after updating the values in the redux store as well.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/run-hot-key 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.2 **(0)** | 36.99 **(0)** | 34.15 **(-0.01)** | 55.72 **(0)**
 :red_circle: | app/client/src/actions/editorActions.ts | 90 **(-10)** | 100 **(0)** | 66.67 **(-33.33)** | 100 **(0)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 67.35 **(0.22)** | 40.48 **(0.12)** | 51.22 **(-0.06)** | 67.28 **(0.23)**
 :red_circle: | app/client/src/reducers/uiReducers/editorReducer.tsx | 22.22 **(-0.64)** | 0 **(0)** | 22.22 **(-1.78)** | 22.22 **(-0.64)**
 :red_circle: | app/client/src/sagas/ActionSagas.ts | 12.06 **(-0.07)** | 0.76 **(-0.01)** | 4 **(-0.17)** | 14.85 **(-0.1)**
 :red_circle: | app/client/src/sagas/ActionExecution/PluginActionSaga.ts | 17.6 **(-0.14)** | 2.73 **(-0.02)** | 10.53 **(0)** | 20.19 **(-0.09)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :red_circle: | app/client/src/selectors/editorSelectors.tsx | 74.72 **(-0.14)** | 66.67 **(0.52)** | 57.63 **(-0.99)** | 71.53 **(-0.3)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/constants/ReduxActionConstants.tsx [hunk 1]
```diff
@@ -248,6 +248,8 @@ export const ReduxActionTypes = {
   FETCH_ACTIONS_SUCCESS: "FETCH_ACTIONS_SUCCESS",
   UPDATE_ACTION_INIT: "UPDATE_ACTION_INIT",
   UPDATE_ACTION_SUCCESS: "UPDATE_ACTION_SUCCESS",
+  /* This action constants is for identifying the starting of the updates */
```
Comment (ayushpahwa) [inline app/client/src/constants/ReduxActionConstants.tsx:251]: Please change this name to something like PRE_INIT or something that mentions that it is called before INIT

File: app/client/src/actions/queryPaneActions.ts [hunk 2]
```diff
@@ -27,3 +27,9 @@ export const changeQuery = (
     payload: { id, newQuery },
   };
 };
+
+/* This action when executed updates the status of isSaving query to true for an actionId */
+export const updateActionStarted = (payload: ReduxAction<{ id: string }>) => ({
+  type: ReduxActionTypes.UPDATE_ACTION_STARTED,
+  payload: payload,
+});
```
Comment (ayushpahwa) [inline app/client/src/actions/queryPaneActions.ts:35]: Please update the name here too.

**Review by ayushpahwa — CHANGES_REQUESTED** [review]
Please update the description to give an overview of the fix.

**Review by hetunandu — DISMISSED** [review]
LGTM. Just update the names

**Review by rashmi-rai — APPROVED** [review]

**Review by hetunandu — CHANGES_REQUESTED** [review]
Can't name the action INIT_UPDATE_ACTION because UPDATE_ACTION_INIT already exists.

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 3]
```diff
@@ -397,6 +400,25 @@ class CodeEditor extends Component<Props, State> {
     CodeEditor.updateMarkings(this.editor, this.props.marking);
   };
 
+  handleDebouncedChange = _.debounce(this.handleChange, 600);
+
+  startHandleChange = (instance?: any, changeObj?: any) => {
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:405]: rename to startChange

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 4]
```diff
@@ -397,6 +400,25 @@ class CodeEditor extends Component<Props, State> {
...
+      entityInformation.entityId
+    ) {
+      const id: string = entityInformation.entityId as string;
+      const payload: any = {
+        id,
+      };
+      /* This action updates the status of the api to isStarting true so that any
+      shortcut commands do not execute before updating the input in the redux store */
+      this.props.preparingForUpdateAction(payload);
+    }
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:418]: This needs to be generic. Dispatch an action called: `ENTITY_UPDATE_STARTED` with entity type and id. We should not handle actions separately in a generic component

File: app/client/src/reducers/uiReducers/queryPaneReducer.ts [hunk 5]
```diff
@@ -54,6 +54,17 @@ const queryPaneReducer = createReducer(initialState, {
     ...state,
     lastUsed: action.payload.id,
   }),
+  /* This action updates the status of the queryPane isSaving for an actionId of the application in the redux store */
+  [ReduxActionTypes.PREPARING_UPDATE_ACTION]: (
```
Comment (hetunandu) [inline app/client/src/reducers/uiReducers/queryPaneReducer.ts:58]: Use `ENTITY_UPDATE_STARTED` here and look for your entity type and id. Only update if it is an action

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Review by hetunandu — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Comment by AmanAgarwal041** [comment]
ok-to-test sha=896025a

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**